service/storage_service.cc

```diff
@@ -3766,7 +3766,7 @@ storage_service::get_splits(const sstring& ks_name, const sstring& cf_name, rang
         std::sort(range_tokens.begin(), range_tokens.end());
         std::move(range_tokens.begin(), range_tokens.end(), std::back_inserter(tokens));
     }
-    tokens.push_back(std::move(unwrapped[unwrapped.size() - 1].end().value_or(range_type::bound(dht::maximum_token()))).value());
+    tokens.push_back(std::move(unwrapped[unwrapped.size() - 1].end().value_or(range_type::bound(dht::greatest_token()))).value());
 
     // split_count should be much smaller than number of key samples, to avoid huge sampling error
```

Comment:
The caller is in thrift:

```c++
    void describe_splits_ex(thrift_fn::function<void(std::vector<CfSplit>  const& _return)> cob, thrift_fn::function<void(::apache::thrift::TDelayedException* _throw)> exn_cob, const std::string& cfName, const std::string& start_token, const std::string& end_token, const int32_t keys_per_split) {
        service_permit permit = obtain_permit();
        with_cob(std::move(cob), std::move(exn_cob), [&]{
            dht::token_range_vector ranges;
            auto tstart = start_token.empty() ? dht::minimum_token() : dht::token::from_sstring(sstring(start_token));
            auto tend = end_token.empty() ? dht::maximum_token() : dht::token::from_sstring(sstring(end_token));
            range<dht::token> r({{ std::move(tstart), false }}, {{ std::move(tend), true }});
```

It's used by clients to attempt to direct queries at the correct node.

Reply:
Why are you pointing this out? Do you mean that this particular place isn't very important?

Reply:
You wrote "I don't understand the full significance of this change." so I'm pointing you at the user.